Data/Sequence/Internal.hs

```diff
@@ -4554,32 +4554,35 @@ popMin cmp = State unrollPQ'
 -- elements, converts a 'Seq' to a 'PQueue'.
 toPQ :: (e -> e -> Ordering) -> Seq e -> Maybe (PQueue e)
 toPQ cmp' (Seq xs') = toPQTree cmp' (\(Elem a) -> PQueue a Nil) xs'
+
+toPQTree :: (b -> b -> Ordering) -> (a -> PQueue b) -> FingerTree a -> Maybe (PQueue b)
+toPQTree _ _ EmptyT = Nothing
+toPQTree _ f (Single xs) = Just (f xs)
+toPQTree cmp f (Deep _ pr m sf) =
+    Just (maybe (pr' <+> sf') ((pr' <+> sf') <+>) m')
   where
-    toPQTree :: (b -> b -> Ordering) -> (a -> PQueue b) -> FingerTree a -> Maybe (PQueue b)
-    toPQTree _ _ EmptyT = Nothing
-    toPQTree _ f (Single xs) = Just (f xs)
-    toPQTree cmp f (Deep _ pr m sf) =
-        Just (maybe (pr' <+> sf') ((pr' <+> sf') <+>) m')
-      where
-        pr' = toPQDigit cmp f pr
-        sf' = toPQDigit cmp f sf
-        m' = toPQTree cmp (toPQNode cmp f) m
-        (<+>) = mergePQ cmp
-    toPQDigit :: (b -> b -> Ordering) -> (a -> PQueue b) -> Digit a -> PQueue b
-    toPQDigit cmp f dig =
-        case dig of
-            One a -> f a
-            Two a b -> f a <+> f b
-            Three a b c -> f a <+> f b <+> f c
-            Four a b c d -> (f a <+> f b) <+> (f c <+> f d)
-      where
-        (<+>) = mergePQ cmp
-    toPQNode cmp f node =
-        case node of
-            Node2 _ a b -> f a <+> f b
-            Node3 _ a b c -> f a <+> f b <+> f c
-      where
-        (<+>) = mergePQ cmp
+    pr' = toPQDigit cmp f pr
+    sf' = toPQDigit cmp f sf
+    m' = toPQTree cmp (toPQNode cmp f) m
+    (<+>) = mergePQ cmp
+
+toPQDigit :: (b -> b -> Ordering) -> (a -> PQueue b) -> Digit a -> PQueue b
```

Comment:
Do you really need them top-level, or is asking a signature for `toPQNode` sufficient?